src/gravel/controllers/nodes/systemdisk.py

```diff
@@ -0,0 +1,255 @@
+    source: str
+    dest: str
+
+
+def get_mounts() -> List[MountEntry]:
+    proc: Path = Path("/proc/mounts")
+    assert proc.exists()
+
+    lst: List[MountEntry] = []
+    with proc.open(mode="r", encoding="utf-8") as f:
+        for line in f.readlines():
+            fields: List[str] = line.split(" ")
+            if len(fields) < 2:
+                continue
+            lst.append(MountEntry(source=fields[0], dest=fields[1]))
+    return lst
+
+
+class SystemDisk:
+
+    _gstate: GlobalState
+    _overlaydirs: Dict[str, str] = {
+        "etc": "/etc",
+        "logs": "/var/log",
+        "etcd": "/var/lib/etcd",
+        "containers": "/var/lib/containers"
+    }
+    _bindmounts: Dict[str, str] = {
+        "ceph": "/var/lib/ceph"
+    }
+
+    def __init__(self, gstate: GlobalState) -> None:
+        self._gstate = gstate
+
+    @property
+    def mounted(self):
+        mounts: List[MountEntry] = get_mounts()
+        for entry in mounts:
+            if entry.source == "/dev/mapper/aquarium-systemdisk" and \
+               entry.dest == "/aquarium":
+                return True
+        return False
+
+    async def lvm(self, args: str) -> None:
+        cmd: List[str] = ["lvm"] + shlex.split(args)
+        retcode, out, err = await aqr_run_cmd(cmd)
+
+        logger.debug(f"lvm stdout: {out}")
+        logger.debug(f"lvm stderr: {err}")
+
+        if retcode != 0:
+            raise LVMError(msg=err)
+
+    async def create(self, devicestr: str) -> None:
+
+        logger.debug(f"prepare system disk: {devicestr}")
+        inventory: Optional[NodeInfoModel] = self._gstate.inventory.latest
+        assert inventory is not None
+
+        device: Optional[VolumeDeviceModel] = \
+            next((d for d in inventory.disks if d.path == devicestr), None)
+
+        # ascertain whether we can use this device
+        if not device:
+            raise UnknownDeviceError(f"device {devicestr} not known")
+        elif not device.available:
+            raise UnavailableDeviceError(f"device {devicestr} is not available")
+
+        # create partitions
+        logger.debug(f"prepare system disk: device: {device}")
+        devpath = device.path
+
+        def _create_overlay_dir(path: Path, dirname: str) -> None:
+            assert path.exists()
+            dirpath: Path = path.joinpath(dirname)
+            assert not dirpath.exists()
+            dirpath.mkdir()
+            dirpath.joinpath("overlay").mkdir()
+            dirpath.joinpath("temp").mkdir()
+
+        try:
+            # create lvm volume
+            await self.lvm(f"pvcreate {devpath}")
+            await self.lvm(f"vgcreate aquarium {devpath} --addtag @aquarium")
+            await self.lvm("lvcreate -l 100%FREE -n systemdisk aquarium")
+
+            lvmdev: Path = Path("/dev/mapper/aquarium-systemdisk")
+            assert lvmdev.exists()
+
+            # format with xfs
+            await aqr_run_cmd(shlex.split(f"mkfs.xfs -m bigtime=1 {lvmdev}"))
```

Comment:
(nit) this is probably simple enough for our environment, but we could also look at using something like pyparted. (Not sure about LVM support though)

Reply:
This was the only way I found to be able to run lvm that isn't N-years old without evidence of being maintained. I'll look into pyparted, but given it seems to be solely for libparted I'm inclined with going with the current approach.